PIPReader/src/main/java/it/cnr/iit/ucs/pipreader/PIPReader.java

```diff
@@ -271,6 +273,23 @@ private final void setFilePath( String filePath ) {
         Reject.ifBlank( this.filePath );
     }
 
+    private void logOperation( String... strings ) {

```

Comment:
Every JournalingInterface object could wrap a Optional<Journal> inside and if it is present it will be used.
In this way we won't need to replicate this function in any component that needs a journal.
if no journal is configured in the component it can be fatal or not but it is decided in the initialization of the actual component.

Reply:
done that, here things are different because there is the formatting of the string